app/controllers/interviews_controller.rb

```diff
@@ -0,0 +1,177 @@
+class InterviewsController < ApplicationController
+    before_action :set_interview, only: [:show, :edit, :update, :destroy]
+
+    def index
+      @interviews = Interview.all
+    end
+
+    def show
+    end
+
+    def new
+      @interview = Interview.new
+      @interview.interviews_users.build(role: 0).build_user
+      @interview.interviews_users.build(role: 1).build_user
+
+      logger.debug "AAAAA #{@interview.users.last.inspect}"
+    end
+
+  def edit
+  end
+
+  def create    
+    @interview = Interview.new(:interview_date => interview_params[:interview_date], :start_time => interview_params[:start_time], :end_time => interview_params[:end_time])
+
+    interviewer_params = interview_params[:interviews_users_attributes]["0"][:users]
+    candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+    @interviewer = User.where(:email => interviewer_params[:email])
+    @candidate = User.where(:email => candidate_params[:email])
+
+    if !(@interviewer.empty?)
+        @interview.users << @interviewer.first
+    else
+        @interview.users << User.new(interviewer_params)
+    end
+    @interview.interviews_users.last.role = 0
+
+    if !(@candidate.empty?)
+        @interview.users << @candidate.first
+    else
+        @interview.users << User.new(candidate_params)
+    end
+    @interview.interviews_users.last.role = 1
+
+    respond_to do |format|
+      if @interview.save
+        welcome
+        interview_create_mail
+        interview_reminder_mail
+        format.html { redirect_to interviews_path, notice: 'Interview schedule was successfully created.'}
+        format.json { render :show, status: :created, location: @interview }
+      else
+        format.html { render :new }
+        format.json { render json: @interview.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  def update
+    interviewer_params = interview_params[:interviews_users_attributes]["0"][:users]
+    candidate_params = interview_params[:interviews_users_attributes]["1"][:users]
+
+    @interviewer = User.where(:email => interviewer_params[:email])
+    @candidate = User.where(:email => candidate_params[:email])
+
+    if !(@interviewer.empty?)
+        if @interviewer.update(interviewer_params)
+            @interview.users << @interviewer
+        else
+            @interview.errors.add(:users, "invalid")
+        end
+    else
+        @interview.users << User.new(interviewer_params)
+    end
+
+    if !(@candidate.empty?)
+        @candidate.update(candidate_params)
+        @interview.users << @candidate
+    else
+        @interview.users << User.new(candidate_params)
+    end
```

Comment:
This portion is repeatative. You can move this to a before_action method on your controller